c10/xpu/test/impl/XPUCachingAllocatorTest.cpp

```diff
@@ -31,13 +33,46 @@ TEST(XPUCachingAllocatorTest, DeviceCachingAllocate) {
   void* tmp = sycl::aligned_alloc_device(
       512, _10mb, c10::xpu::get_raw_device(0), c10::xpu::get_device_context());
   void* ptr1 = c10::xpu::XPUCachingAllocator::raw_alloc(_10mb);
-  // ptr0 and ptr1 should be on the same block.
+  // ptr0 and ptr1 should be on the same block, and ptr1 is the next block of
+  // ptr0, like [ptr0, ptr1]. So the offset of their pointers should equal to
+  // ptr0's size (10M).
```

Comment:
Can you elaborate the assumption? Do you assume that the block size is >=10M and also there is no previous allocation before the test? Can you also explain why the assumption holds?

Reply:
rich comments